src/tile/models/tileManager.ts

```diff
@@ -0,0 +1,197 @@
+        (successResponse: Buffer) => {
+          resolve(successResponse);
+        },
+        (errorResponse: Error) => {
+          reject(errorResponse);
+        }
+      );
+    });
+  }
+
+  private renderImage(zoom: number, lon: number, lat: number, successCallback: (buffer: Buffer) => void, errorCallback: (err: Error) => void): void {
+    this.renderersPool.acquire((error: Error | null, renderer: Map) => {
+      if (error) {
+        errorCallback(error);
+      }
+
+      const params = this.getRenderParams(zoom, lon, lat);
+
+      renderer.render(params, (error, buffer) => {
+        this.renderersPool.release(renderer);
+
+        if (error) {
+          return errorCallback(error);
+        }
+
+        this.removeSemiTransparentOutlinesOnBuffer(buffer);
+
+        sharp(buffer, {
+          raw: {
+            width: params.width as number,
+            height: params.height as number,
+            channels: NUM_OF_CHANNELS_IN_BUFFER,
+          },
+        })
+          .png({ adaptiveFiltering: false })
+          .toBuffer({ resolveWithObject: false })
+          .then(successCallback)
+          .catch(errorCallback);
+      });
+    });
+  }
+
+  private getRenderParams(zoom: number, lon: number, lat: number): RenderParams {
+    const { tileSize, zoom: zoomSettings } = this.application;
+    return {
+      zoom: Math.max(zoomSettings.min, tileSize === DEFAULT_TILE_SIZE ? zoom - 1 : zoom),
+      center: [lon, lat],
+      width: tileSize,
+      height: tileSize,
+    };
+  }
+
+  private removeSemiTransparentOutlinesOnBuffer(buffer: Buffer): void {
+    for (let i = 0; i < buffer.length; i += NUM_OF_CHANNELS_IN_BUFFER) {
+      const alpha = buffer[i + BufferChannel.ALPHA];
+      if (alpha === MAX_ALPHA_VALUE) {
+        continue;
+      }
+      const norm = alpha / MAX_ALPHA_VALUE;
+
+      // if fully-transparent change to black, otherwise normalize the color
+      if (alpha === 0) {
+        buffer[i + BufferChannel.RED] = BLACK;
+        buffer[i + BufferChannel.GREEN] = BLACK;
+        buffer[i + BufferChannel.BLUE] = BLACK;
+      } else {
+        buffer[i + BufferChannel.RED] /= norm;
+        buffer[i + BufferChannel.GREEN] /= norm;
+        buffer[i + BufferChannel.BLUE] /= norm;
+      }
+    }
+  }
+
+  private createPool(ratio: number, minResources: number, maxResources: number): Pool<Map> {
+    const createRenderer = (ratio: number, createCallback: (err: Error | null, renderer: Map) => void): void => {
+      const renderer = new Map({ ...this.renderHandler.renderOptions, ratio });
+      renderer.load(this.global.styleContent);
+      createCallback(null, renderer);
+    };
+    return new Pool({
+      min: minResources,
+      max: maxResources,
+      create: createRenderer.bind(null, ratio),
+      destroy: (renderer: Map): void => {
+        renderer.release();
+      },
+    });
+  }
+
+  private createShutdownTimeout(): void {
+    if (testConfig.enabled && poolInactivityTimeout == undefined) {
```

Comment:
I couldn't find a proper way to close the pool while testing, `jest` got stuck between tests because it detected open handlers.
We should consider the use of another pooling library in the future.

Reply:
problem solved using `fastify` onClose hook